src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,108 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+}) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const currentPage = Number(searchParams.get('page'));
+  const perPage = Number(searchParams.get('perPage'));
+
+  const totalPages = Math.ceil(total / perPage);
+  const pageButtons = getNumbers(1, totalPages)
+    .map(n => (
+      <li
+        key={n}
+        className={cn(
+          'page-item',
+          // eslint-disable-next-line quote-props
+          { 'active': n === currentPage },
+        )}
+      >
+        <a
+          data-cy="pageLink"
+          className="page-link"
+          href={`#${n}`}
+          onClick={() => {
+            if (n !== currentPage) {
+              searchParams.set('page', String(n));
+              setSearchParams(searchParams);
+            }
+          }}
```

Comment:
If you want to use search params with pagination move the logic to  separate function instead of writing in the event

Reply:
Done.